src/puppet.ts

```diff
@@ -855,7 +855,15 @@ export abstract class Puppet extends EventEmitter {
 
     const roomPayloadList = await Promise.all(
```

Comment:
Please add a `.filter` to the `roomPayloadList` so that we will not get a `undefined` in the data.